src/libDirectoryService/DSBlockPreProcessing.cpp

```diff
@@ -222,14 +230,41 @@ void DirectoryService::InjectPoWForDSNode(VectorOfPoWSoln& sortedPoWSolns,
     return;
   }
 
+  if (numOfRemovedMembers > numOfProposedDSMembers) {
```

Comment:
while the previous implementation is `>`, I think it may make more sense it we use `!=` here

Reply:
No, `numOfRemovedMembers` must be strictly less than or equals to `numOfProposedDSMembers` because `numOfRemovedMembers` contains the members to be removed for non-performance only, not expiry.

Reply:
Ok then this will make sense.